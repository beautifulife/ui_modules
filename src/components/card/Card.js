import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import './Card.scss';
import Cover from './Cover';
import Information from './Information';
import Rating from './Rating';

const Card = ({
  cardAuthor,
  cardCrossOut,
  cardHilight,
  cardImgSrc,
  cardLabel,
  cardRating,
  cardRatingDesc,
  cardTitle,
  cardType = 'vertical'
}) => {
  return (
    <Fragment>
      {cardType === 'vertical' ? (
        <div className="Card vertical">
          <Cover imgSrc={cardImgSrc} label={cardTitle} type={cardType} />
          <Information
            crossOut={cardCrossOut}
            hilight={cardHilight}
            label={cardLabel}
            title={cardTitle}
          />
          {cardRating && (
            <Rating
              rating={cardRating}
              ratingDesc={cardRatingDesc}
              type={cardType}
            />
          )}
        </div>
      ) : (
        <div className="Card horizontal">
          <div className="cell">
            <Cover imgSrc={cardImgSrc} label={cardTitle} type={cardType} />
          </div>
          <div className="cell">
            <Rating
              author={cardAuthor}
              rating={cardRating}
              ratingDesc={cardRatingDesc}
              type={cardType}
            />
          </div>
        </div>
      )}
    </Fragment>
  );
};

Card.propTypes = {
  cardAuthor: PropTypes.string,
  cardCrossOut: PropTypes.string,
  cardHilight: PropTypes.string,
  cardImgSrc: PropTypes.string,
  cardLabel: PropTypes.string,
  cardRating: PropTypes.number,
  cardRatingDesc: PropTypes.string,
  cardTitle: PropTypes.string,
  cardType: PropTypes.string
};

Card.defaultProps = {
  cardAuthor: '',
  cardCrossOut: '',
  cardHilight: '',
  cardImgSrc: '',
  cardLabel: '',
  cardRating: -1,
  cardRatingDesc: '',
  cardTitle: '',
  cardType: 'vertical'
};

export default Card;
