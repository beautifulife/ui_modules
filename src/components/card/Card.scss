.Card {
  display: inline-block;
  vertical-align: top;
  width: 200px;
  margin: 0 15px;
  border: 1px solid #eee;
  border-radius: 3px;

  &.horizontal {
    display: table;
    width: 400px;
    height: 150px;
  }
}
