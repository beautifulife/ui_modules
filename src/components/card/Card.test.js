import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

describe('Card', () => {
  describe('about render', () => {
    const wrapper = shallow(<Card />);

    it('compare with snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('render horizontal when cardType is horizontal', () => {
      wrapper.setProps({ cardType: 'horizontal' });
      expect(wrapper.find('.Card.horizontal').length).toBe(1);
    });
  });
});
