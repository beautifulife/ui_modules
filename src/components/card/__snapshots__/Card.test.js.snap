// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card about render compare with snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    cardAuthor=""
    cardCrossOut=""
    cardHilight=""
    cardImgSrc=""
    cardLabel=""
    cardRating={-1}
    cardRatingDesc=""
    cardTitle=""
    cardType="vertical"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="Card vertical"
        >
          <Cover
            imgSrc=""
            label=""
            type="vertical"
          />
          <Information
            crossOut=""
            hilight=""
            label=""
            title=""
          />
          <Rating
            author=""
            rating={-1}
            ratingDesc=""
            type="vertical"
          />
        </div>,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Cover
              imgSrc=""
              label=""
              type="vertical"
            />,
            <Information
              crossOut=""
              hilight=""
              label=""
              title=""
            />,
            <Rating
              author=""
              rating={-1}
              ratingDesc=""
              type="vertical"
            />,
          ],
          "className": "Card vertical",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "imgSrc": "",
              "label": "",
              "type": "vertical",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "crossOut": "",
              "hilight": "",
              "label": "",
              "title": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "author": "",
              "rating": -1,
              "ratingDesc": "",
              "type": "vertical",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      false,
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="Card vertical"
          >
            <Cover
              imgSrc=""
              label=""
              type="vertical"
            />
            <Information
              crossOut=""
              hilight=""
              label=""
              title=""
            />
            <Rating
              author=""
              rating={-1}
              ratingDesc=""
              type="vertical"
            />
          </div>,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Cover
                imgSrc=""
                label=""
                type="vertical"
              />,
              <Information
                crossOut=""
                hilight=""
                label=""
                title=""
              />,
              <Rating
                author=""
                rating={-1}
                ratingDesc=""
                type="vertical"
              />,
            ],
            "className": "Card vertical",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "imgSrc": "",
                "label": "",
                "type": "vertical",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "crossOut": "",
                "hilight": "",
                "label": "",
                "title": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "author": "",
                "rating": -1,
                "ratingDesc": "",
                "type": "vertical",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        false,
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
